bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDVertex.java

```diff
@@ -284,6 +280,16 @@ public DFDVertex copy(Map<DFDVertex, DFDVertex> mapping) {
         return new DFDVertex(this.referencedElement, copiedPinDFDVertexMap, new HashMap<>(this.pinFlowMap));
     }
 
+    @Override
+    public UUID getUniqueIdentifier() {
+        StringBuilder uuidString = new StringBuilder();
+        uuidString.append(this.getReferencedElement().getId());
+        for (var previousElement : this.getPreviousElements()) {
+            uuidString.append(previousElement.getUniqueIdentifier());
+        }
+        return UUID.nameUUIDFromBytes(uuidString.toString().getBytes(StandardCharsets.UTF_8));
+    }
```

Comment:
This is not enough to uniquely identify the vertex.
Assume a DFD like this:
```
               B
Z - A <           > D
               C
```
               
This would mean 2 partial TFG's ZABD and ZACD (assuming both flows land in the same pin of D).

It is impossible to differentiate between the Z/A vertices in the different partial flow graphs with the information contained in the Class. This is always the case if they share the same sequence of previous vertices.
It's fine if there is a different sequence of previous vertices (use of different pin, loop, etc.).

However if they share the same sequence they are equal in everything (previous vertices sequence/ data/node characteristics), so maybe it's fine if they share an ID?

Otherwise, the only difference is in which Partial TFG they are, so the ID needs to be derived from the partial flow graph.

Reply:
Or am I misunderstanding your point?

Reply:
Maybe my diagram wasn't clear
![image](https://github.com/user-attachments/assets/d7c2e1d7-dd9b-4f20-8677-5956f0dd24a1)
This would for example create 2 User vertices in two different partial tfg's. Their parameters would be equal

Reply:
![image](https://github.com/user-attachments/assets/8a408c86-1e2d-4ca0-ba6e-ab774efc316c)

In this case, we would have 2 TFGs. The ID for User vertices would still be equal and by extent also the id for the function node.

I think the ID needs to come from either the Partial flow graph containing the vertex we're trying to get the ID for., or by using UUID.randomUUID in the abstractVertex

Reply:
The whole point of the UUID is to have a unique, **not random** ID for elements that stays the same during runs of the analysis.
Therefore, I would further revise the approach to include the ID of the outgoing Port as well.

In the Example, to get the UUID of the Function Node, it would first calculate the UUID of the previous vertex and pass its Node ID and the ID of the sending Port to it. Therefore, the UUIDs of the two Vertices that reference User would be different, as the sending port is different.

I know this solution is pretty complicated, but is still better that really not being able to reference a vertex between runs consistently.

Reply:
I've implemented this in 0d14062e9f4db90c375da6d5d704978035331b54

Reply:
Nothing against creating the type 3 uuids from some meaningful properties from the nodes or something, but if we do not need those properties to be encoded there, I opt for using the random uuid functionality. 
Looking into how EcoreUtils does this, they seem to mainly use the timestamp to generate a type 2 uuid. As huge as the type 4 UUIDs that are generated are, I do not see a problem.

This would also make the code simpler as it just needs to be implemented in AbstractVertex.

Reply:
pinDFDVertexMap only maps the input Pins to the previous vertices. A vertex holds no information about its successor. Therefore in this case Function would grap user to compute its ID. As Users pinDFDVertexMap is empty the user vertices would be indistinguishable. 

If you don't want random ID's or to use the partial TFG they are residing in, you need to make the ID depended on the Ids of the incoming flows and then hand it to its predecessor. The incoming flow is the only part in which the vertices differ.